fakebook/src/app/models/post.ts

```diff
@@ -0,0 +1,11 @@
+import { User } from './user';
+import { Comment } from './comment';
+
+export interface Post {
+  id: number;
+  content: string;
+  user: User;
+  pictureUrl: string | undefined;
+  createdAt: Date;
```

Comment:
This `createdAt` should be a union type of `Date | undefined`,  unless you can guarantee that `Post#createdAt` will always be defined.